django_socio_grpc/mixins.py

```diff
@@ -228,19 +235,34 @@ def get_default_message(model_name, fields="__all__"):
         }
 
 
+def _get_partial_update_request(service):
+    class PartialUpdateRequest(service.get_serializer_class()):
+        _partial_update_fields = serializers.ListField(child=serializers.CharField())
+
+    return PartialUpdateRequest
+
+
 class PartialUpdateModelMixin(GRPCActionMixin, abstract=True):
-    @grpc_action(request=SelfSerializer, response=SelfSerializer)
+    @grpc_action(
+        request=FnPlaceholder(_get_partial_update_request),
+        request_name=StrTemplatePlaceholder(
+            f"{{}}PartialUpdate{REQUEST_SUFFIX}", get_serializer_base_name
+        ),
+        response=SelfSerializer,
+    )
     def PartialUpdate(self, request, context):
         """
         Partial update a model instance.
 
-        The request have to include a field corresponding to
-        ``lookup_request_field`` and you need to explicitly set the fields that
-        you want to update.  If an object is updated this returns a proto
-        message of ``serializer.Meta.proto_class``.
+        Performs a partial update on the given `_partial_update_fields`.
         """
+
+        content = message_to_dict(request)
+
+        message = {k: v for k, v in content.items() if k in request._partial_update_fields}
+
         instance = self.get_object()
-        serializer = self.get_serializer(instance, message=request, partial=True)
+        serializer = self.get_serializer(instance, data=message, partial=True)
```

Comment:
add a comment above this line to explain that we need to use data because it's converted to a dict instead of an grpc message object. To avoid someone replace it in the future